container/state.go

```diff
@@ -385,6 +385,17 @@ func (s *State) IsDead() bool {
 // closes the internal waitRemove channel to unblock callers waiting for a
 // container to be removed.
 func (s *State) SetRemoved() {
+	s.ErrorMsg = ""
+	s.Lock()
+	close(s.waitRemove) // Unblock those waiting on remove.
+	s.Unlock()
+}
+
+// SetRemovalError is to be called in case a container remove failed.
+// It sets an error and closes the internal waitRemove channel to unblock
+// callers waiting for the container to be removed.
+func (s *State) SetRemovalError(err error) {
+	s.SetError(err)
 	s.Lock()
 	close(s.waitRemove) // Unblock those waiting on remove.
```

Comment:
A problem still exists where another API request could come along after this to try to remove the container again. If it succeeds and `SetRemoved()` is called then it will panic because we've closed this channel again which was already closed.

It should be okay to just create a new `s.waitRemove` channel after closing the original one.

Reply:
Sure. I have updated the patch to reopen the channel